### ISSUE TITLE
Provide template variable for ISO date

### ISSUE THREAD

**Issue by mlncn** [swentel/indieweb#361]
For some use cases, a node template variable of the ISO8601 machine-readable date helps (to manually provide the "datetime" property, say like this:

```
<time class="dt-published" datetime="{{ iso_datetime }}">{{ date }}</time>
```

There may be other ways to achieve this more in line with the overall architecture of the IndieWeb module.  But this fits with the `{{ date }}` variable Drupal already provides for great flexibility in marking up dt-published information.

This is in a sense a follow-up to #354 --- once we have our ability to have multiple or remote authors through a userreference field, an external link field, or (the [Drutopia](http://docs.drutopia.org/en/latest/end-user-documentation.html#step-by-step-instructions-for-adding-content) use case) content references to a person content type, the next thing we need to complete a valid h-entry is marking up the post date, as the author information enhanced by IndieWeb module or the metadata is hardcoded to a single author, even after #360 lands.  (And i don't think what #1 covers does this.)

## Background

In theory Drupal core's date formatter functionality could do all of this.  In practice, something weird happens (and escaping tons of letters in order to output HTML tags and class names is pretty darn ugly).

Here's the something weird that happens, for the record— only the last two digits coming after the final colon are printed:

```
<\t\i\m\e \c\l\a\s\s="\d\t-p\ub\l\i\s\h\e\d" \d\a\t\e\t\i\m\e="c">Y F d</\t\i\m\e>
```

And in the preview shown after save, it looks great!

```
<time class="dt-published" datetime="2018-12-22T15:05:42-05:00">2018 December 22</time>
```

But in the source of something actually using it... it comes out like this:

```
<time class="dt-published" datetime="00">2015 November 11</time>
```

And this:

```
<\t\i\m\e \c\l\a\s\s="\d\t-p\ub\l\i\s\h\e\d" \d\a\t\e\t\i\m\e="Y-m-d h:m:i">Y F j</\t\i\m\e>
```

which previews like this:

```
<time class="dt-published" datetime="2018-12-22 03:12:16">2018 December 22</time>
```

ends up like this:

```
<time class="dt-published" datetime="49">2015 November 11</time>
```

**Comment by mlncn** [comment]
So i can't get a merge request that doesn't inexplicably drag in all my commits from #355 even though those were merged into master, but the change here is trivial— and probably it should be done differently.

Right now **iso_datetime** is only available as a node template variable if "dt-published, p-name, u-author and u-url in a hidden span element on nodes." is checked on the Microformats configuration page (so that `post_metadata` equals true).  Maybe it should be a separate option?  Or there should be a set of variables added if a separate option is checked?  Happy to create a fresh pull request with any direction that meets the use case outline above.

Thanks!

**Comment by mlncn** [comment]
@swentel what is your workflow with github to Drupal.org?  Github claims that it's master version and Drupal.org's 8.x-1.x "are entirely different commit histories"!

https://github.com/swentel/indieweb/compare/master...agaric:8.x-1.x?expand=1

**Comment by mlncn** [comment]
Anyhow grabbed and applied the changes to the GitHub repo rather than the Drupal.org repo.

Now that i think about it, i think you were interpreting my suggestion as a single option to expose multiple template variables— but i don't know what the other variables would be?  Happy to change the option name to be more generic and future-proof, and to provide any other suggested variables now also, if that was the idea.

Our web site is so close to launching with at least as much IndieWeb goodness as we hacked in 3+ years ago!  Thank you @swentel !

**Comment by swentel** [comment]
> what is your workflow with github to Drupal.org

Oh yes sorry about this one. I don't know exactly why I simply didn't start using multiple remotes (I probably had some (weird) good reason, but never documented it) On my local machine, I copy from my github branch to the drupal remote in a totally different directory. I'm planning to move everything to d.o once the integration with gitlab is finished though.

> i think you were interpreting my suggestion as a single option to expose multiple template variables

Hmm yes, I guess I thought it would make sense to provide variables for u-author and u-url, however, all those variables are already available in one form or another, so starting with this one is fine.

**Comment by swentel** [comment]
merged and pushed to d.o as well, thanks!

**Comment by mlncn** [comment]
Yay, thank you!

And, as long as you have a system  :-)

And sorry about the typos!

/me turns in his journalism degree.
